generic/build

```diff
@@ -30,23 +30,20 @@
 set -e
 
 die() {
-	local m="$1"
-
-	echo "FATAL: ${m}" >&2
+	echo "FATAL: $1" >&2
 	exit 1
 }
 
 get_full_path() {
-	local d="$1"
-	( cd "${d}" 2> /dev/null && pwd )
+	( cd "$1" 2> /dev/null && pwd )
 }
 
 empty_ifelse() {
-	local v="$1"
-	local t="$2"
-	local f="$3"
+	[ -z "$1" ] && echo "$2" || echo "$3"
+}
 
-	[ -z "${v}" ] && echo "$2" || echo "$3"
+contains() {
+    [ "${1#*$2}" != "$1" ] && echo "$3" || echo "$4"
```

Comment:
This is an interesting construct and does seem to work as advertised. Do you happen to have a link where this is explained?

Reply:
I mean the ```${1#*$2}``` part of course.

Reply:
somewhere in Bourne shell documentation, I cannot recall now.

I tried to keep it as much away from bash as possible

Reply:
It seemed to work fine in Dash which tries to be POSIX-compliant. So this is definitely not specific to Bash.

Reply:
I tested on /bin/sh under freebsd, which I suppose is Bourne shell